File: PN31.cpp


#include<iostream>
using namespace std;

class Number
{
    private:
        int iNo;       

    public:
        int iCnt = 0;
		
        void Accept()       
        {
            cout<<"Enter the value : "<<endl;
            cin>>this->iNo;
        }

        void DisplayEven()
		{
			int iDigit = 0;
			
			
			if(iNo < 0)
			{
				iNo = -iNo;
			}
			while(iNo > 0)
			{
				iDigit = iNo % 10;
				if((iDigit % 2) == 0)
				{
				   cout<<iDigit<<endl;
				}   
				iNo = iNo /10;
			}
			
		}
};

int main()
{
    Number nobj;
	int iRet = 0;
	


    nobj.Accept();

	nobj.DisplayEven();
	


    return 0;
}

File: PN31.java
import java.lang.*;
import java.util.*;

class Number
{
    private int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter number : ");
        this.iNo = sobj.nextInt();
    }

    public void Display()
    {
        System.out.println("Value is : "+this.iNo);
    }

    public void Factorial()
    {
        int i = 0;
		
		for(i = 1; i<= iNo/2;i++)
		{
			if((i%2) == 0)
			{
				System.out.println(i);
			}
		}
			
           
    }
}

class PN31
{
    public static void main(String b[])
    {
            Number nobj = new Number();
            int iRet = 0;

            nobj.Accept();
            nobj.Display();

            nobj.Factorial();
            
    }
}